subgraphs/aave-v2-forks/protocols/radiant/src/rewards.ts

```diff
@@ -0,0 +1,134 @@
+import { Address, BigDecimal, BigInt, log } from "@graphprotocol/graph-ts";
+import {
+  RDNT_WETH_Uniswap_Pair,
+  RWETH_ADDRESS,
+  REWARD_TOKEN_ADDRESS,
+  rTOKEN_DECIMALS,
+} from "./constants";
+import {
+  BIGDECIMAL_ZERO,
+  DEFAULT_DECIMALS,
+  exponentToBigDecimal,
+  RewardTokenType,
+  SECONDS_PER_DAY,
+} from "../../../src/constants";
+import { ReserveDataUpdated } from "../../../generated/LendingPool/LendingPool";
+import { RToken } from "../../../generated/LendingPool/RToken";
+import { getOrCreateRewardToken } from "../../../src/helpers";
+import { Market } from "../../../generated/schema";
+import { ChefIncentivesController } from "../../../generated/LendingPool/ChefIncentivesController";
+import { UniswapV2Pair } from "../../../generated/LendingPool/UniswapV2Pair";
+
+export function updateMarketRewards(
+  event: ReserveDataUpdated,
+  market: Market,
+  rTokenContract: RToken
+): void {
+  const TWELVE_HOURS = BigInt.fromI32(SECONDS_PER_DAY / 2);
+  if (
+    market._lastRewardsUpdated &&
+    !event.block.timestamp.minus(market._lastRewardsUpdated!).gt(TWELVE_HOURS)
+  ) {
+    return;
+  }
+
+  let tryIncentiveController = rTokenContract.try_getIncentivesController();
+  if (tryIncentiveController.reverted) {
+    return;
+  }
+
+  let incentiveController = ChefIncentivesController.bind(
+    tryIncentiveController.value
+  );
+  let tryDepPoolInfo = incentiveController.try_poolInfo(
+    Address.fromString(market.outputToken!)
+  );
+  let tryBorPoolInfo = incentiveController.try_poolInfo(
+    Address.fromString(market.vToken!)
+  );
+  let tryTotalAllocPoint = incentiveController.try_totalAllocPoint();
+  let tryTotalRewardsPerSecond = incentiveController.try_rewardsPerSecond();
+
+  if (
```

Comment:
I think these should be ORs. If not all are reverted it won't skip the function and could try to read value on a reverted contract call.

Reply:
Oh, good catch! Changing that 🚀